src/Adapter/Product/CommandHandler/UpdateProductCustomizationFieldsHandler.php

```diff
@@ -0,0 +1,162 @@
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\CommandHandler\DeleteCustomizationFieldHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\CommandHandler\UpdateCustomizationFieldHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\CommandHandler\UpdateProductCustomizationFieldsHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\CustomizationField;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Query\GetProductCustomizationFields;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\QueryHandler\GetProductCustomizationFieldsHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+/**
+ * Handles @var UpdateProductCustomizationFieldsCommand using legacy object model
+ */
+class UpdateProductCustomizationFieldsHandler extends AbstractCustomizationFieldHandler implements UpdateProductCustomizationFieldsHandlerInterface
+{
+    /**
+     * @var AddCustomizationFieldHandlerInterface
+     */
+    private $addCustomizationFieldHandler;
+
+    /**
+     * @var UpdateCustomizationFieldHandlerInterface
+     */
+    private $updateCustomizationFieldHandler;
+
+    /**
+     * @var DeleteCustomizationFieldHandlerInterface
+     */
+    private $deleteCustomizationFieldHandler;
+
+    /**
+     * @var GetProductCustomizationFieldsHandlerInterface
+     */
+    private $getProductCustomizationFieldsHandler;
+
+    /**
+     * @param AddCustomizationFieldHandlerInterface $addCustomizationFieldHandler
+     * @param UpdateCustomizationFieldHandlerInterface $updateCustomizationFieldHandler
+     * @param DeleteCustomizationFieldHandlerInterface $deleteCustomizationFieldHandler
+     * @param GetProductCustomizationFieldsHandlerInterface $getProductCustomizationFieldsHandler
+     */
+    public function __construct(
+        AddCustomizationFieldHandlerInterface $addCustomizationFieldHandler,
+        UpdateCustomizationFieldHandlerInterface $updateCustomizationFieldHandler,
+        DeleteCustomizationFieldHandlerInterface $deleteCustomizationFieldHandler,
+        GetProductCustomizationFieldsHandlerInterface $getProductCustomizationFieldsHandler
+    ) {
+        $this->addCustomizationFieldHandler = $addCustomizationFieldHandler;
+        $this->updateCustomizationFieldHandler = $updateCustomizationFieldHandler;
+        $this->deleteCustomizationFieldHandler = $deleteCustomizationFieldHandler;
+        $this->getProductCustomizationFieldsHandler = $getProductCustomizationFieldsHandler;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Creates, updates or deletes customization fields depending on differences of existing and provided fields
+     */
+    public function handle(UpdateProductCustomizationFieldsCommand $command): array
```

Comment:
😮 so you have to "find out" yourself what must be added, updated or deleted ? legacy code did not do that ?

Reply:
It was all done in ProductWrapper.
But I just found out I've missed product "customizable" property. I'll check that.

Reply:
I JUST MERGE NOW THEN

Reply:
as @jolelievre found out, the Product->customizable can have 3 possible values as it seems.
* no custom fields (uncostomizable) = 0
* at least one custom field (no required) = 1
* at least one custom field, and at least one required = 2

So Ill need to Introduce some VO to handle this. (`Domain\Product\ValueObject\ProductCustomability`?)
Moreover handlers were not updating count of `product->uploadable_files` & `product->text_fields` which apparently reflects how many customization fields exists based on its type.

**So there are still left to add following actions in this pr** 

in `AddCustomizationFieldHandler`:
* After field is added, update product->customizable field
* After field is added, increment uploadable_files || text_fields count depending on field type.

and in `DeleteCustomizationFieldHandler`:
* After field is deleted, check if it wasnt the last field and update product->customizable prop accordingly.
* After field is deleted, check existing fields count and update uploadable_files || text_fields accordingly.